### ISSUE TITLE
Mails rejected by gmx.de (and others) if FROM contains special chars

### ISSUE THREAD

**Issue by Lineflyer** [osTicket/osTicket#6436]
### Prerequisites

* [x] Can you reproduce the problem in a fresh installation of the "develop" branch?
* [x] Do you have any errors in the PHP error log, or javascript console?
* [x] Did you check the [osTicket forums](https://forum.osticket.com/)?
* [x] Did you [perform a cursory search](https://github.com/osTicket/osTicket/issues) to see if your bug or enhancement is already reported?

### Description
I kept having issues to send SMTP mails to the provider GMX, which I initially described in https://github.com/osTicket/osTicket/issues/6295#issuecomment-1276701568

As the error description was quite generic I started playing around with several different SMTP accounts, where some worked and some not.

It turned out, that messages start being rejected if the FROM name contain special characters. As soon as I remove them it seems working smoothly and the remote server accepts and delivers the mail.

### Steps to Reproduce

1. Configure Admin-Panel, E-Mails, E-Mails
2. Select or create a valid IMAP/SMTP configuration and set the FROM field to ```a:bcd support```
3. Save the mail configuration
4. Use Diagnosis to send an outgoing mail to a a valid recipient `@gmx.de`

**Expected behavior:** 
Mail will be delivered
Note:
Sticking to the above example, it works as soon as I e.g. change the FROM field to ```abcd support```

**Actual behavior:**
Mail is rejected by target server
Error provided, see: https://github.com/osTicket/osTicket/issues/6295#issuecomment-1276701568

```
host mx01.emig.gmx.net [212.227.17.5]
    SMTP error from remote mail server after end of data:
    554-Transaction failed
    554-Reject due to policy restrictions.
    554 For explanation visit https://www.gmx.net/mail/senderguidelines?ip=193.95.241.239&c=hi
```
https://www.gmx.net/mail/senderguidelines?ip=193.95.241.239&c=hi

### Versions
osTicket V1.17.2

![image](https://user-images.githubusercontent.com/949669/212544124-5d5be2b8-e631-4d58-916c-dd45872f7c95.png)

**Comment by Lineflyer** [comment]
As this looks to me like a clear bug...does anyone have an idea how to patch/fix it?
Its easy to avoid (as decribed above), however with leaving out the `:` in the FROM we are loosing our brand name.

**Comment by JediKev** [comment]
@Lineflyer 

Interesting, I'm using latest `develop` (new release coming soon) and when I set my Email Name to `Support:Name` or `Support : Name` and send via Diagnostic the email is properly recieved and the name is not UTF-8 encoded. Maybe it's my mail provider but looking at the raw email it looks like `From: "Support:Name" <email@domain.tld>` (properly quoted and not UTF-8 encoded).

Cheers.

**Comment by JediKev** [comment]
@Lineflyer 

Do you have raw headers for an email that's being rejected? This will help us figure out what's going on as I'm unable to replicate this currently. (I also don't have a german email account to test with either so)

Cheers.

**Comment by Lineflyer** [comment]
Taken from the returned mail eml attachement with the original mail in it.
Anonymized the recipient, rest is not modified.

```
Return-path: <support@cgeo.org>
Received: from [2001:1640:5::3:d5] (helo=w205.goneo.de)
    by gorenc.org with esmtpsa (TLS1.3:ECDHE_RSA_AES_256_GCM_SHA384:256)
    (Exim 4.92)
    (envelope-from <support@cgeo.org>)
    id 1pH26Q-0008No-M6
    for xxxx@gmx.de; Sun, 15 Jan 2023 13:27:10 +0100
Date: Sun, 15 Jan 2023 12:27:10 +0000
Message-ID: <B3drOb6-cfemu-AAAAAB1DAAAzFgAATRhRfJ9A-support@cgeo.org>
From: =?utf-8?Q?c:geo=20Support?= <support@cgeo.org>
Subject: =?UTF-8?Q?Re:=20Test=20(plain=20text)=20[#806834]?=
In-Reply-To: <ffdc2189-aae0-cd6d-d479-00831f94c4d7@gmx.de>
References: <ffdc2189-aae0-cd6d-d479-00831f94c4d7@gmx.de>
X-Mailer: =?UTF-8?Q?osTicket=20Mailer?=
To: =?utf-8?Q?xxxx?= <xxxx@gmx.de>
MIME-Version: 1.0
Content-Type: multipart/alternative;
    boundary="=_43a1f0fb12279b1bb67b40cb0fa490e6"
```

**Comment by JediKev** [comment]
@Lineflyer 

Are you using Sendmail or SMTP? It sounds like you're using Sendmail.

Cheers.

**Comment by Lineflyer** [comment]
Using SMTP (see above)

**Comment by JediKev** [comment]
@Lineflyer 

Are you sure SMTP is actually working though? Can you check your Admin Panel > Dashboard > System Logs to see if you see any Mailer Errors? Also, you can check the Sent folder of the email (`support@`) to see if the emails are there, if not then it's not using SMTP.

Cheers.

**Comment by Lineflyer** [comment]
Yes, SMTP is working and no errors in system logs.
It has to be SMTP as the SENDMAIL of my provider would not accept a foreign FROM mail address. 

osticket is running on my webspace (goneo.de) and fetching via IMAP and sending via SMTP (gorenc.org) . No other way.

**Comment by JediKev** [comment]
@Lineflyer 

Okay, just ruling everything I can out first as I'm not able to replicate this.

Cheers.

**Comment by Lineflyer** [comment]
I am no encoding expert...but could it be a malformed UTF8 encoding of the `c:geo support`?
As said...if I remove the `:` GMX is accepting the mail.

**Comment by JediKev** [comment]
@Lineflyer 

Yes, however, I'm trying to figure out how that's happening for you as I mentioned mine is properly quoted and not encoded (ie. `From: "Support:Name" <email@domain.tld>`).

Cheers.

**Comment by Lineflyer** [comment]
Interesting, let me check some older headers while still running osticket 1.16.

**Comment by Lineflyer** [comment]
Indeed, before upgrade to 1.17 it was using ```From: "c:geo Support" <support@cgeo.org>```

Edit:
Also X-MAILER and SUBJECT have not been UTF-8 encoded in 1.16

If I remove the `:` in 1.17  its also still UTF-8 encoded:
```From: =?utf-8?Q?cgeo=20Support?= <support@cgeo.org>```

Edit: Could this be mail server configuration issue or is the encoding decided by the mailer (=osticket)?
